stable_baselines3/common/callbacks.py

```diff
@@ -675,15 +676,43 @@ def __init__(self) -> None:
         self.pbar = None
 
     def _on_training_start(self) -> None:
-        # Initialize progress bar
+
+        # Get the total number of timesteps to do this training session
+        total = self.locals["total_timesteps"]
+
         # Remove timesteps that were done in previous training sessions
-        self.pbar = tqdm(total=self.locals["total_timesteps"] - self.model.num_timesteps)
+        total -= self.model.num_timesteps
+
+        # Round timesteps up-to a multiple of the algorithm's batch size
```

Comment:
I'm not sure about that part.
As stated in https://github.com/DLR-RM/stable-baselines3/issues/1150, it is normal that some algorithms do more steps than expected.
The progress bar is only a rough indicator of how long a training will take and `total_timesteps` is a lower bound, so I think I would prefer to update the documentation (to make it clearer) rather than having some hacky code in there (that cover most cases but not all cases as you have probably noticed).

Reply:
@qgallouedec what do you think?

Reply:
I agree. Under-reporting should be fixed, but not over-reporting because there is no such thing as over-reporting.